components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java

```diff
@@ -0,0 +1,263 @@
+                    siddhiDefinitions.add(siddhiDefinition);
+                }
+            }
+
+            //Get Table defintions with @store annotations
+            for (Map.Entry<String, TableDefinition> entry : siddhiApp.getTableDefinitionMap().entrySet()) {
+                if (entry.getValue().toString().contains(STORE_ANNOTATION)) {
+                    SiddhiDefinition siddhiDefinition = new SiddhiDefinition(entry.getKey(), entry.getValue()
+                            .toString(), TABLE, entry.getValue().getAttributeList());
+                    siddhiDefinitions.add(siddhiDefinition);
+                }
+            }
+
+            //Get Window defintions with @store annotations
+            for (Map.Entry<String, WindowDefinition> entry : siddhiApp.getWindowDefinitionMap().entrySet()) {
+                if (entry.getValue().toString().contains(STORE_ANNOTATION)) {
+                    SiddhiDefinition siddhiDefinition = new SiddhiDefinition(entry.getKey(), entry.getValue()
+                            .toString(), WINDOW, entry.getValue().getAttributeList());
+                    siddhiDefinitions.add(siddhiDefinition);
+                }
+            }
+            Collections.sort(siddhiDefinitions);
+            String jsonString = new Gson().toJson(siddhiDefinitions);
+            return Response.ok().entity(jsonString).build();
+        } catch (feign.RetryableException e) {
+            log.warn("Unable to reach the worker " + workerURI, e);
+        } catch (Exception e) {
+            log.warn("Error occured while getting response from worker " + workerURI, e);
+        }
+        return Response.ok(new ApiResponseMessage(ApiResponseMessage.ERROR, "Error occured while getting the" +
+                " response from " + workerURI)).build();
+    }
+
+    @Override
+    public Response getAllSiddhiApps(Request request) {
+        if (getUserName(request) != null && !getPermissionProvider().hasPermission(getUserName(request), new
+                Permission(PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi " +
+                    "Apps").build();
+        }
+        ConfigReader cf = new ConfigReader();
+        String username = cf.getUserName();
+        String password = cf.getPassword();
+        ArrayList<String> workerList =  cf.getWorkerList(); //list of worker urls
+
+        //Map to store siddhi apps from all the workers
+        Map<String, String> siddhiApps = new ConcurrentHashMap<>();
+
+        workerList.parallelStream().forEach(worker -> {
+            try {
+                feign.Response workerResponse = WorkerServiceFactory.getWorkerHttpsClient(
+                        PROTOCOL + worker, username, password).getSiddhiAppNames();
+                if (workerResponse != null && workerResponse.status() == 200) {
```

Comment:
Is only Response.Ok from worker run times handled?

Reply:
Fixed in a8ae221